pkg/cmd/roachtest/spec/cluster_spec.go

```diff
@@ -466,26 +469,48 @@ func (s *ClusterSpec) RoachprodOpts(
 		}
 	}
 
+	var workloadMachineType string
+	var err error
+	switch cloud {
+	case AWS:
+		workloadMachineType, _, err = SelectAWSMachineType(s.WorkloadNodeCPUs, s.Mem, preferLocalSSD && s.VolumeSize == 0, requestedArch)
```

Comment:
Technically, `SelectXXXMachineType` is not well-defined for the case when `s.WorkloadNodeCPUs == 0`.